src/app.ts

```diff
@@ -13,14 +14,50 @@ interface TemperatureSummary {
   average: number
 }
 
+const actualTemperatures: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
   // add here your code
+  actualTemperatures.push(...readings)
+}
+
+function averageTemperature(readings: TemperatureReading[]) {
+  const result = readings.reduce(
+    (sum, actualVal) => sum + actualVal.temperature,
+    0,
+  )
+  return result / readings.length
+}
+
+function sameDay(date1: Date, date2: Date) {
+  if (
+    date1.getDate() === date2.getDate() &&
+    date1.getMonth() === date2.getMonth() &&
+    date1.getMonth() === date2.getMonth()
+  )
+    return true
+  return false
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
   //add here your code
-  return null
+  const todayTemperatures: TemperatureReading[] = actualTemperatures.filter(
+    (value) => {
```

Comment:
Use more descriptive names for your variables. Instead of `value`, you could use `temperature`